crates/symex_lib/src/any.rs

```diff
@@ -1,46 +1,56 @@
 use super::symbolic;
-use std::mem::MaybeUninit;
 
+/// Provides the [`any`] operation on the implementor
+/// ---
+///
+/// The any operation should return a new [`symbolic`] value with no prior [`assumptions`](crate::assume) placed on the value
+///
+/// ### Example implementation
+/// ```no_run
+/// impl Any for u8 {
+///     /// Generates a symbolic value a without any assumption of the value.
+///     fn any() -> Self{
+///         unsafe{
+///             let mut a = core::mem::MaybeUninit::uninit();
+///             symex_lib::symbolic(&mut a);
+///             a.assume_init()
+///         }
+///     }
+/// }
+/// ```
 pub trait Any {
     fn any() -> Self;
 }
 
+/// Creates a new [`symbolic`] value with no prior assumptions.
+/// ---
+///
+/// The symbolic value returned should have no [`assumptions`](crate::assume`) placed on it after the function returns.
 #[inline(never)]
 pub fn any<T: Any>() -> T {
     T::any()
 }
 
-macro_rules! blanket_impl {
-    ( $type: ty) => {
-        impl Any for $type {
-            #[inline(always)]
-            fn any() -> Self {
-                internal_any()
+/// Implements the [`Any`] trait for the specified types.
+///
+/// The blanket implementation provides a method [`any`] that creates a [`uninit`](core::mem::MaybeUninit) value
+/// that is converted to a [`symbolic`] value which is then assumed to be initialized to make the compiler happy.
+macro_rules! impl_primitive {
+    ( $($type: ty),+) => {
+        $(
+            impl Any for $type {
+                /// Generates a new symbolic value from uninitialized memory
+                #[inline(always)]
+                fn any() -> Self {
+                    unsafe {
+                        let mut a = core::mem::MaybeUninit::uninit();
+                        symbolic(&mut a);
+                        a.assume_init()
+                    }
+                }
             }
-        }
+        )+
     };
 }
 
-blanket_impl!(bool);
-
-blanket_impl!(u8);
-blanket_impl!(u16);
-blanket_impl!(u32);
-blanket_impl!(u64);
-blanket_impl!(u128);
-blanket_impl!(usize);
-
-blanket_impl!(i8);
-blanket_impl!(i16);
-blanket_impl!(i32);
-blanket_impl!(i64);
-blanket_impl!(i128);
-blanket_impl!(isize);
-
-fn internal_any<T: Any>() -> T {
```

Comment:
For these cases I think it's fine to inline it. However, it *may* have to be re-used when actually doing `bool` [correctly](https://doc.rust-lang.org/nomicon/what-unsafe-does.html). `Any` is also something that probably should be implemented for some standard types, but again, not sure if this will be re-used or not.

But anyway, not sure how correct this function is [0], [1], [2], if you want you can have a look at it. The only two requirements are really (1) construct a value without `Default` and (2) don't let the optimizer go crazy. Then the docs would be perfect for describing why are aren't doing UB that would maybe cause miscompiles.

\[0]: <https://doc.rust-lang.org/core/ptr/fn.read_volatile.html#safety>
[1]: <https://doc.rust-lang.org/core/mem/union.MaybeUninit.html>
[2]: <https://doc.rust-lang.org/nomicon/unchecked-uninit.html>